### PR TITLE
📚 docs: open details section when linking items on it

### PR DESCRIPTION
**Pull request by letiescanciano**
## What
Closes [#8754](https://github.com/airbytehq/airbyte-internal-issues/issues/8754)
Hash linking (e.g https://docs.airbyte.com/integrations/sources/stripe#rate-limiting ) to items inside `<details>` elements does not cause `<details>` to be opened by default.
These changes improves UX so if the pathname contains a hash that it's inside a `<details>` element, it will open the element.

This works also for the title of the section (the element just before  `<details>` )

## How
- Implementing a custom component for `<details>` and setting it to open based on  current url.

## Can this PR be safely reverted and rolled back?
- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: docusaurus/src/components/Details.module.css
Comment (lmossman) [inline docusaurus/src/components/Details.module.css:3]: One different I noticed between this and the original docusaurus `details` is that the docusaurus one would animate a "slide down" of the content when opening, and "slide up" when closing, but this custom one just immediately opens or closes the full content.

I don't think this is a huge deal if it is hard to get working right, but just wanted to point it out

File: docusaurus/src/theme/MDXComponents/index.js
Comment (lmossman) [inline docusaurus/src/theme/MDXComponents/index.js:24]: I'm surprised that this is all that is needed to replace the default `details` component!

Is that something specific to docusaurus, such that you just need to have a component named the same (but with a capitalized first letter)?

File: docusaurus/src/components/Details.jsx
Comment (lmossman) [inline docusaurus/src/components/Details.jsx:6]: I tried testing with a nested `<details>` tag. 

While opening a new tab with a hash of a nested header worked, clicking on another header and then clicking on the header in that nested `<details>` section causes the entire parent details section to be collapsed. See recording below:

https://github.com/user-attachments/assets/6202181a-1750-46be-bd2a-d6ed62e54c1a

Ideally this would work as well, but doesn't necessarily need to block this merge.

**Review by lmossman — APPROVED** [review]
Had a few comments, up to you whether to address them before merging or in a follow-up PR. Otherwise this looks good to me, nice!

File: docusaurus/src/theme/MDXComponents/index.js
Comment (letiescanciano) [reply docusaurus/src/theme/MDXComponents/index.js:24]: Me too! In fact, I created the component and use it on the .md files with <Details> instead of <details>, but I realized that other elements in the page were also changing without changing the component used.

I thought this will be handy so people don't have to remember using the JSX component instead of the "plain" html one. But if you feel it's confusing, I can rename the component so we still have both!

Comment (letiescanciano) [reply docusaurus/src/theme/MDXComponents/index.js:24]: I think is actually a Markdown thing. It seems that both `<Details>` and `<details>` will render become `<details>`.

Eg: <Details> will be rendered automatically as <details></details> </Details>


This was easy to spot on docusaurus by logging `MDXComponents from "@theme-original/MDXComponents"`;  and our CustomComponents export.
I'm actually overwriting the definition of `<Details>`  by using the same name, which it turns out it overwrites both.

I think this is useful in this case, because this is the default behaviour that we want for this component, but let me know if you would prefer to have a separate component!

https://www.loom.com/share/f1842ecd722a43bdb3e878ac0888949b?sid=a1ed795d-70b5-499c-ad6c-ea12dc7fbb27

File: docusaurus/src/components/Details.jsx
Comment (letiescanciano) [reply docusaurus/src/components/Details.jsx]: Nice catch! I'll create a new issue and prioritize it with Natalie :)

File: docusaurus/src/components/Details.module.css
Comment (letiescanciano) [reply docusaurus/src/components/Details.module.css:3]: I tried to copy the behaviour but can't make it work. I'll add this in a new issue as a nice-to-have to unblock this one :)

File: docusaurus/src/theme/MDXComponents/index.js
Comment (lmossman) [reply docusaurus/src/theme/MDXComponents/index.js:24]: No this makes sense, I am pleasantly surprised that the overriding is this easy to configure 👍

**Pull request merged**
